src/main/java/com/flightstats/hub/dao/aws/DynamoWebhookDao.java

```diff
@@ -74,13 +76,23 @@ public Webhook get(String name) {
             if (result.getItem() == null) {
                 return null;
             }
-            return mapItem(result.getItem());
+            return tryMapItem(result.getItem())
+                    .orElseThrow(() -> new ResourceNotFoundException("Unable to read webhook config from dynamo " + name));
```

Comment:
The flow never hits path orElseThrow(()). tryMapItem returns empty instance or value.
Also in this method there are 3 return statements. can this consolidated to single exit?

Reply:
I did some refactoring to reduce the number of returns and to remove the confusing `orElseThrow` bit.